dfm/train.py

```diff
@@ -0,0 +1,182 @@
+        return tmp
+
+
+@dataclass
+class ModelArguments(object):
+    model_type: str
+    model_name: str
+
+    @classmethod
+    def from_yaml(cls, config_path: str):
+        file = read_yaml(config_path)
+        keys = cls.__dataclass_fields__.keys()
+        yaml_data = file["model_arguments"]
+        normal_yaml_data = {key: yaml_data[key] for key in yaml_data if key in keys}
+        anormal_yaml_data = {
+            key: yaml_data[key] for key in yaml_data if key not in keys
+        }
+        tmp = cls(**normal_yaml_data)
+        for anormal_key in anormal_yaml_data:
+            setattr(tmp, anormal_key, anormal_yaml_data[anormal_key])
+        return tmp
+
+
+class DFMTrainer:
+    """class for training a Danish Foundation Model."""
+
+    def __init__(
+        self,
+        pretraining_config_path: str,
+    ) -> None:
+        """
+        Training function with Weights & Biases logging.
+
+        Args:
+            model_name (str): Path to a model.
+            pretraining_config_path (str): Path to a pretraining config.
+            datasets (list(str)): List of dataset names from the Hugging Face hub.
+            interleave_probabilities (List[float]): how to weight datasets in interleave
+        """
+        # Configuration arguments
+        self.pretraining_config_path = pretraining_config_path
+        self.training_args = read_yaml(self.pretraining_config_path)[
+            "training_arguments"
+        ]
+        self.data_args = DataArguments.from_yaml(self.pretraining_config_path)
+        self.model_args = ModelArguments.from_yaml(self.pretraining_config_path)
+
+        self.dataset_names = self.data_args.dataset_names
+        self.interleave = self.data_args.interleave
+        self.interleave_probabilities = self.data_args.interleave_probabilities
+
+        self.model_type = self.model_args.model_type
+        self.model_name = self.model_args.model_name
+
+        if self.model_type not in MODEL_TYPES:
+            raise ValueError(f"Invalid model type. Choose one from: {MODEL_TYPES}")
+
+    def train(
+        self,
+    ):
+
+        HF_config = AutoConfig.from_pretrained(self.model_name)
+        tokenizer = AutoTokenizer.from_pretrained(
+            self.model_name, return_special_tokens_mask=True
+        )
+
+        # Load and preprocess datasets
+        datasets = [dfm_load_dataset(d) for d in self.dataset_names]
+        datasets = [
+            preprocess_dataset(d, tokenizer=tokenizer, num_proc=self.data_args.num_proc)
+            for d in datasets
+        ]
+        train_datasets = [d["train"] for d in datasets]
+        eval_datasets = [d["val"] for d in datasets]
+
+        # Interleave
+        train_datasets = interleave_datasets(
+            train_datasets, probabilities=self.interleave_probabilities
+        )
+        eval_datasets = interleave_datasets(
+            eval_datasets, probabilities=self.interleave_probabilities
+        )
```

Comment:
I feel like this line should simply be:
```
datasets = dfm_load_dataset(dataset_configs)
train = datasets["train"]
# ...
```

Reply:
I agree - but until we have the full streaming functionality and preprocessing of datasets, this is a workaround for training.

Reply:
Perfect, I assumed it was a training workaround as well, just wanted to align on the intended design.